### PR TITLE
Update readme, bump min_ansible_version

### PR DESCRIPTION
**Pull request by justinc1**
The changes were requested in "Ansible Partner Announcements - May 2024"

### CODE REVIEW

**Review by shoriminimoe — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -3,28 +3,72 @@
 The Ansible Collection for Scale Computing HyperCore ([HyperCore](https://www.scalecomputing.com/sc-hypercore))
 a variety of Ansible content to help automate the management of Scale Computing HyperCore products.
 
+Documentation is available at https://scalecomputing.github.io/HyperCoreAnsibleCollection/.
 
+# Requirements
 <!--start requires_ansible-->
-## Ansible version compatibility
+## Ansible version requirements
```
Comment (domendobnikar) [inline README.md:10]: Nice change, more python like wording 🥇

File: README.md [hunk 2]
```diff
@@ -3,28 +3,72 @@
...
 
 This collection has been tested against following HyperCore cluster versions:
 - v9.1.14.208456
 - v9.2.13.211102
 - v9.3.5.212852
 - v9.4.0.213268
 
+# Installation
+
+Before using this collection, you need to install it with the Ansible Galaxy command-line tool:
```
Comment (domendobnikar) [inline README.md:32]: Maybe something like this:
"To use this collection, first install it using the Ansible Galaxy command-line tool:"

**Review by domendobnikar — APPROVED** [review]
Looks good, thank you 👍

File: README.md (hunk 2 above)
Comment (justinc1) [reply README.md:32]: I will leave this as is - I verbatim copied from RH template.

**Pull request merged**